dpki/src/keystore.rs

```diff
@@ -0,0 +1,330 @@
+use crate::{
+    key_bundle::KeyBundle,
+    keypair::generate_random_sign_keypair,
+    seed::{generate_random_seed_buf, IndexedSeed, RootSeed, SeedContext, SeedTrait, SeedType},
+    utils, SEED_SIZE,
+};
+use holochain_core_types::{
+    agent::Base32,
+    cas::content::Address,
+    error::{HcResult, HolochainError},
+    signature::Signature,
+};
+
+use holochain_sodium::{kdf, secbuf::SecBuf};
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+pub enum Secret {
+    Key(KeyBundle),
+    RootSeed(RootSeed),
+    IndexedSeed(IndexedSeed),
+}
+
+#[allow(dead_code)]
+struct Keystore {
+    keys: HashMap<String, Arc<Mutex<Secret>>>,
```

Comment:
Do you think it would be worth adding in a stub for requesting the passphrase (it can return a test string or something), then either stubs or real implementations of the encryption / decryption / disk persistence?

I don't know how realistic interacting with a `HashMap<String, Arc<Mutex<Secret>>>` is for the target solution.

Reply:
The passphrase doesn't happen at the keystore level.  There is a single keystore is per-instance.  The keystore manager does the passphrase.